### ISSUE TITLE
Replace `admin:admin` default credentials

### ISSUE THREAD

**Issue by setiah** [opensearch-project/OpenSearch#1650]
**Is your feature request related to a problem? Please describe.**
The default `admin:admin` credential is too weak and a sure-fire way to get hacked. Though the documentation recommends users to change the default passwords, it can be easily missed. Such weak defaults can be very easily exploited and lead to massive data leaks (remember [Equifax attack](https://www.forbes.com/sites/kateoflahertyuk/2019/10/20/equifax-lawsuit-reveals-terrible-security-practices-at-time-of-2017-breach/?sh=446983d93d38)?). 

The second gap is, in the existing distribution, there's no default password policy to ensure stronger passwords. It needs to be configured explicitly by users. That is why such weak credentials are able to slip through the cracks.

**Describe the solution you'd like**
Remove the default passwords provided in `internal_users.yml`. Instead, have the `admin` user setup passwords for internal users once the OpenSearch process is up. These passwords can be set up post-installation using a REST API and `admin` certificates (for authentication). Setting password for `admin` user would look something like -
```
PATCH _opendistro/_security/api/internalusers/admin --key kirk-key.pem --cert kirk.pem --cacert root-ca.pem -d'
{
    "password": "quCn52_8\*N8"
}
'
```

Another way is that the OpenSearch distribution provide tools like an OpenSearch shell client to make it easy for admin user to configure all users. This shell tool can be extended for use for more administrative operations in future. 
 
In addition to this, we should also add a default password policy to only allow passwords that match a certain strength criteria. Obv, these policies are overridable depending on the user/enterprise needs. Having a strong default policy will protect against insecure password configurations.

**Describe alternatives you've considered**

**Additional context**
This is related to https://github.com/opensearch-project/OpenSearch/issues/1618

**Comment by dredwilliams** [comment]
Agreed -- the inability to change the admin password in particular without wiping the internal user database is a problem.

**Comment by AmiStrn** [comment]
It seems like this is assuming use of the security plugin. It should be optional to use the security plugin. 

I believe this should be mobed to the security plugin repo here - https://github.com/opensearch-project/security

**Comment by setiah** [comment]
@CEHENKLE I don't have permissions to transfer issues to the security plugin repo. Could you pls do that?

**Comment by PhilipMay** [comment]
This is a super important security issue. Can you please prioritize it?

PS: it seems like there is something that could handle this - but it is 1. not documented and 2. I am not sure how and if it even works.

https://github.com/opensearch-project/OpenSearch/blob/0ba0e7cc26060f964fcbf6ee45bae53b3a9941d0/distribution/docker/src/docker/bin/docker-entrypoint.sh#L58
